### ISSUE TITLE
E522 is too picky

### ISSUE THREAD

**Issue by TotalVerb** [tonyhffong/Lint.jl#140]
```julia
julia> lintstr("""
       macro int_str(x::String)
           parse(BigInt, x)
       end
       """)
1-element Array{Lint.LintMessage,1}:
 none:1 E522 x::String: macro arguments can only be Symbol/Expr
```

**Comment by TotalVerb** [comment]
And... who would write `macro blah(x::Type)` if they did not mean that `x` should be of type `Type`? I think E522 should be thrown out.

**Comment by Michael-Klassen** [comment]
Lint often tries to be too smart for it's own good. I would not be against cutting down on some of the unnecessary errors/warning, or at least reduce their severity. `Error` should only be used on things guaranteed to fail.

**Comment by tonyhffong** [comment]
don't you mean 
```julia
function int_str( x::String )
 #...
end
```

I do not know of an instance where ```macro blah( x::Type )``` is correct.

**Comment by TotalVerb** [comment]
No, I don't mean that. The string macro syntax `int"x"` is actually a `@int_str "x"` macrocall. Example:

```julia
macro int_str(x::String)
    parse(BigInt, x)
end

int"1000"
```
